src/charm.py

```diff
@@ -191,14 +195,16 @@ def _sync_profiles(self):
                     str(self.config.get(ISTIO_PRINCIPAL_KEY)),
                 )
             except ApiError as e:
+                msg = "An unexpected ApiError happened. Please look at the logs for more details"
```

Comment:
nitpick: I think "occurred" is a more fitting word here
```suggestion
                msg = "An unexpected ApiError occurred. Please look at the charm's logs for more details."
```